clippy_lints/src/uninformative_asserts.rs

```diff
@@ -0,0 +1,109 @@
+use crate::utils::span_lint_and_help;
+use if_chain::if_chain;
+use rustc_ast::ast::{Item, MacArgs, MacCall};
+use rustc_ast::token;
+use rustc_ast::tokenstream::TokenTree;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+
+declare_clippy_lint! {
+    /// **What it does:**
+    /// Lint {debug_}assert{_eq,_ne}! without a custom panic message.
+    ///
+    /// **Why is this bad?**
+    /// If the assertion fails, a custom message may make it easier to debug what went wrong.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// # fn some_condition(_x: u8) -> bool { true }
+    /// # fn bar(x: u8) -> u8 { x }
+    /// # let foo = 0u8;
+    /// # let a = 0u8;
+    /// # let b = 0u8;
+    /// assert!(some_condition(foo));
```

Comment:
I think this test should have a `///# #![allow(clippy::uninformative_assert)]` line pretended. Otherwise clippy-checking the doctests would fail.